### DISCUSSION TITLE
VISUAL UI CREATOR for MAUI

### DISCUSSION THREAD

**Discussion by pinkfloyd11** [dotnet/maui#20368] [Ideas]
Sorry folk but......

"visual" studio that not have visual tool for design UI is absurd!!!
Hot reload is a classic 1 step forward and 3 step back

Please ADD UI TOOL as all other language for respect to Visual Basic that have permitted you to be what now you are.
Never forgot past!

Giuseppe

**Comment by dbSoft-SoftwareSolutions** [comment]
+1

**Comment by Pastajello** [comment]
Since MAUI UI is based on native controls from each platform - not drawn by skia or something - what is the look You'd expect?

Lets say take Android, should it show elements from newest Android, API 34 or some oldest supported? Same goes to iOS, which should be shown?

Controls change over versions - toolbar, buttons, labels, entries. If you want to be able to visualize those elements it'd be necessary to map ALL configurations for this visual designer and have a way to change the version of OS you want to look at or the OS itself.
That indeed would be a very powerful tool, but quite hard to make. 

OR you could simply use a device that will render it and reload when you change something.

**Reply by dbSoft-SoftwareSolutions** [reply]
Does flutter or ionic or react native have this tool?

**Reply by pinkfloyd11** [reply]
They are not named Visual flutter or Visual Ionic....
Visual studio have

**Reply by Pastajello** [reply]
But MAUI is not Visual Studio, MAUI is a framework, Visual Studio is an IDE. You should then post your feedback on VS feedback page. I dont see Visual in NET Multi-platform App UI.
